### DISCUSSION TITLE
Extract regionMatrix data from key?

### DISCUSSION THREAD

**Discussion by AndreaRP** [stuart-lab/signac#1336] [Q&A] [answered]
I'm trying to extract the data matrix computed by Signac when running RegionMatrix(). I know I can plot the data with RegionHeatmap(), but I want to plot the data with another package. So, if I created a region matrix like this:

```
my_object <- RegionMatrix(my_object
                        , key = "my_region_matrix"
                        , regions = StringToGRanges(top_cluster_genes$gene)
                        , upstream = 2500, downstream = 2500)
```

How do I go about extracting the data stored within the object under the key "my_region_matrix"? I know it's a pretty basic question, but I've been browsing the returned Seurat object and can't find any assay/matrix with that key name.

**Comment by timoast** [comment, marked as answer]
You can use the `GetAssayData()` function to extract the raw data, eg:

```r
mat <- GetAssayData(object = object, slot = "positionEnrichment")[["my_region_matrix"]]
```

**Reply by AndreaRP** [reply]
That's exactly what I was looking for, thank you!
